From b82f6e4ab98fca46c34f7de881facdac3f464dc7 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 19 Jun 2025 18:00:23 +0200
Subject: [PATCH] arm64: dts: st: change PCIE bus clock reference to use the
 SCMI clock

The PCIE bus clock should be managed by SCMI, therefore, update the
clock reference to use the SCMI clock.

Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I97edb9c7314b0f574d1881054461ced47ed82321
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/468569
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Christian BRUEL <christian.bruel@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2417,7 +2417,7 @@
 				      <0x10000000 0x10000000>;
 				reg-names = "dbi", "dbi2", "atu", "addr_space";
 				st,syscfg = <&syscfg>;
-				clocks = <&rcc CK_BUS_PCIE>;
+				clocks = <&scmi_clk CK_SCMI_ICN_PCIE>;
 				clock-names = "core";
 				resets = <&rcc PCIE_R>;
 				reset-names = "pcie";
@@ -2452,7 +2452,7 @@
 					 <0x42000000 0 0x18000000 0x18000000 0 0x8000000>;
 				dma-ranges = <0x42000000 0x0 0x80000000 0x80000000 0x0 0x80000000>;
 				bus-range = <0x00 0xff>;
-				clocks = <&rcc CK_BUS_PCIE>;
+				clocks = <&scmi_clk CK_SCMI_ICN_PCIE>;
 				clock-names = "core";
 				resets = <&rcc PCIE_R>;
 				reset-names = "pcie";
